picard/ui/options/renaming.py

```diff
@@ -173,22 +191,49 @@ def _example_to_filename(self, file):
                     if s_enabled and s_text:
                         parser = ScriptParser()
                         parser.eval(s_text, file.metadata)
-            filename = file.make_filename(file.filename, file.metadata, settings)
+            filename_before = file.filename
+            filename_after = file.make_filename(filename_before, file.metadata, settings)
             if not settings["move_files"]:
-                return os.path.basename(filename)
-            return filename
+                return os.path.basename(filename_before), os.path.basename(filename_after)
+            return filename_before, filename_after
         except ScriptError:
-            return ""
+            return "", ""
         except TypeError:
-            return ""
+            return "", ""
+
+    def _sample_example_files(self):
+        # Get a new sample of randomly selected /loaded files to use as renaming examples
+        import random
+        max_samples = 10  # pick up to 10 samples
+        if self.tagger.window.selected_objects:
+            # If files/albums/tracks are selected, sample example files from them
+            files = self.tagger.get_files_from_objects(self.tagger.window.selected_objects)
+            length = min(max_samples, len(files))
+            files = [file for file in random.sample(files, k=length)]
+        else:
+            # If files/albums/tracks are not selected, sample example files from the pool of loaded files
+            files = self.tagger.files
+            length = min(max_samples, len(files))
+            files = [files[key] for key in random.sample(files.keys(), k=length)]
+
+        if not files:
+            # If no file has been loaded, use generic examples
+            files = [self.example_1(), self.example_2()]
+        self._sampled_example_files = files
+        self.update_examples()
 
     def update_examples(self):
-        # TODO: Here should be more examples etc.
-        # TODO: Would be nice to show diffs too....
-        example1 = self._example_to_filename(self.example_1())
-        example2 = self._example_to_filename(self.example_2())
-        self.ui.example_filename.setText(example1)
-        self.ui.example_filename_va.setText(example2)
+        self.ui.example_filename_before.clear()
+        self.ui.example_filename_after.clear()
+
+        if not self._sampled_example_files:
+            self._sample_example_files()
+        example_files = self._sampled_example_files
+
+        for example in example_files:
```

Comment:
I would just sort the list by _after_ path, so the list isn't shuffled too much when renewing, and the list is still in ordered track number (title01 before ttile02), this sort may not be correct in all cases, but that's an improvement imho:

```python
        examples = [self._example_to_filename(example) for example in example_files]
        for before, after in sorted(examples, key=lambda x: x[1]):
            self.ui.example_filename_before.addItem(before)
            self.ui.example_filename_after.addItem(after)
```

What do you think?

Reply:
Yes, the random shuffling does not help I think. I think it would be better with the sorting as suggested.

Reply:
I agree. Also, I was testing this and noticed something weird. %title% doesn't seem to be returning the %title% tag, but the filename (at least on the displayed examples). I'm currently looking into it.